design/ceph/block-ceph-delete-if-pool-data.md

```diff
@@ -0,0 +1,146 @@
+
+Given the extra time needed to check pools for their actual usage, this race condition is more
+likely with the optimization but is still relevant even in the simplified case proposed above where
+we only check whether pools exist.
+
+Thought: We could also make it a Rook cleanup policy stop client writes to the Ceph cluster on
+deletion (disable writes by default is safest).
+
+Thought: we can make this optimization an option in our cleanup policy (off by default is safest).
+
+Ceph has different storage types which might be in use, and there are special considerations for
+determining if they are being used.
+
+### CephFilesystems / CephFS
+TODO: Is it safe to delete if there are no user files but there is a directory tree
+created/modified?
+
+TODO: all of this section
+
+### CephObjectStores / RADOS Gateway
+Special consideration for this case is given to additional resources for this case.
+ObjectBucketClaims (OBCs) (with lib-storage-provisioner) and BucketRequests (BRs) (with Container
+Object Storage Interface (COSI)) create a bucket within a CephObjectStore. Because these resources
+exist outside of Rook's scope, the mere existence of these resources should block a CephCluster from
+being deleted.
+
+If we treat existence of a bucket as usage of the object store, we do not need to have special
+handling for blocking CephCluster deletion when OBCs or BRs exist. Therefore, this is the proposal:
+to treat existence of buckets as "intent to use" even if there is no current usage. This will mean
+that CephCluster deletion will block when these resources exist, and CephCluster deletion will be
+allowed when they are removed (as long as the reclaim policy is "Delete").
+
+#### Flow for determining object store usage
+For determining whether object stores in a Ceph cluster hold data, the logic flow below can be used.
+The flow mentions specific `radosgw-admin` commands.
+- `radosgw-admin zonegroup list` to list ZGs
+- For each zonegroup in the list:
+  - `radosgw-admin --rgw-zonegroup=$zonegroup zonegroup get` to get ZG info
+  - For each zone in `zones` (from JSON output):
+    - `radosgw-admin --rgw-zonegroup=$zonegroup --rgw-zone$zone zone get` to get an object store's buckets
+    - `radosgw-admin --rgw-zonegroup=$zonegroup --rgw-zone=$zone bucket list` to get buckets
+    - Ignore a single instance of a bucket named `rook-ceph-bucket-checker-*`
+    - If other buckets exist, assume the store is in-use
+
+##### Corner cases
+1. If a user creates an object store without using CephObjectStore and inside that store creates a
+bucket with a name in the format `rook-ceph-bucket-checker-*`, Rook will not stop cluster deletion.
+2. `.rgw.root` is present in Ceph pool list but not included in info from `radosgw-admin`... is this
+   item ALWAYS the same name?????? (TODO)
+
+##### Background technical information
+Zone name and zonegroup name are usually the same as CephObjectStore name except with new
```

Comment:
I think there is a catch, if this is a multi-site cluster and the admin wants to remove one site (since data are synced to another zone), then it's probably acceptable to go with the deletion.